hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -334,11 +334,12 @@ protected TokenRelation autoAssociate(
                 entitiesConfig.limitTokenAssociations() && numAssociations >= tokensConfig.maxPerAccount(),
                 TOKENS_PER_ACCOUNT_LIMIT_EXCEEDED);
 
-        // TODO: check for unlimited associations flag and include -1 as infinite associations
         final var maxAutoAssociations = account.maxAutoAssociations();
         final var usedAutoAssociations = account.usedAutoAssociations();
-        validateFalse(usedAutoAssociations >= maxAutoAssociations, NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
 
+        if (!entitiesConfig.unlimitedAutoAssociationsEnabled() || account.maxAutoAssociations() != -1) {
```

Comment:
I understand what this if statement is trying to do but if you can leave a comment to clarify your intent, that would be helpful.

Reply:
Even better, as this same check may be needed in some other places, create a method to call here:
```java
    // used in some handler steps, also, needs javadoc...
    protected hasUnlimitedAutoAssociations(@NonNull final Account account) {
        return !(entitiesConfig.unlimitedAutoAssociationsEnabled() && account.maxAutoAssociations() = -1);
    }
```
Then here we would have
```suggestion
        if (hasUnlimitedAutoAssociations(account)) {
```

Reply:
I added a code comment and also applied Joseph's suggestion for creating a method.